src/Adapter/Product/CommandHandler/SetProductCustomizationFieldsHandler.php

```diff
@@ -0,0 +1,190 @@
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
+use PrestaShop\PrestaShop\Core\Exception\CoreException;
+
+/**
+ * Handles @see  SetProductCustomizationFieldsCommand using legacy object model
+ */
+class SetProductCustomizationFieldsHandler extends AbstractCustomizationFieldHandler implements SetProductCustomizationFieldsHandlerInterface
+{
+    /**
+     * @var AddCustomizationFieldHandlerInterface
+     */
+    private $addCustomizationFieldHandler;
+
+    /**
+     * @var CustomizationFieldDeleterInterface
+     */
+    private $customizationFieldDeleter;
+
+    /**
+     * @var CustomizationFieldPersister
+     */
+    private $customizationFieldPersister;
+
+    /**
+     * @var CustomizationFieldProvider
+     */
+    private $customizationFieldProvider;
+
+    /**
+     * @var ProductUpdater
+     */
+    private $productUpdater;
+
+    /**
+     * @param AddCustomizationFieldHandlerInterface $addCustomizationFieldHandler
+     * @param CustomizationFieldPersister $customizationFieldPersister
+     * @param CustomizationFieldDeleterInterface $customizationFieldDeleter
+     * @param CustomizationFieldProvider $customizationFieldProvider
+     * @param ProductUpdater $productUpdater
+     */
+    public function __construct(
+        AddCustomizationFieldHandlerInterface $addCustomizationFieldHandler,
+        CustomizationFieldPersister $customizationFieldPersister,
+        CustomizationFieldDeleterInterface $customizationFieldDeleter,
+        CustomizationFieldProvider $customizationFieldProvider,
+        ProductUpdater $productUpdater
+    ) {
+        $this->addCustomizationFieldHandler = $addCustomizationFieldHandler;
+        $this->customizationFieldDeleter = $customizationFieldDeleter;
+        $this->customizationFieldPersister = $customizationFieldPersister;
+        $this->customizationFieldProvider = $customizationFieldProvider;
+        $this->productUpdater = $productUpdater;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * Creates, updates or deletes customization fields depending on differences of existing and provided fields
+     */
+    public function handle(SetProductCustomizationFieldsCommand $command): array
+    {
+        $deletableFieldIds = $this->getDeletableFieldIds($command);
+
+        foreach ($command->getCustomizationFields() as $customizationFieldDTO) {
+            if ($customizationFieldDTO->getCustomizationFieldId()) {
+                $this->handleUpdate($customizationFieldDTO);
+            } else {
+                $this->handleCreation($command->getProductId(), $customizationFieldDTO);
+            }
+        }
+
+        $this->deleteCustomizationFields($deletableFieldIds);
+        $product = $this->getProduct($command->getProductId());
```

Comment:
Maybe the product could be fetched only once (you also call `$this->getProduct` in `deleteCustomizationFields`) and passed as an argument

Or is it mandatory to get a "fresh" Product after the customization fields have been deleted? In that case maybe you should add a comment here `// Fetch fresh product` or something so that no one thinks of refactoring this part and mess with it